src/Type/Php/MbFunctionsReturnTypeExtension.php

```diff
@@ -74,6 +75,11 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 		$returnType = ParametersAcceptorSelector::selectSingle($functionReflection->getVariants())->getReturnType();
 		$positionEncodingParam = $this->encodingPositionMap[$functionReflection->getName()];
 
+		// php8-stubs define a regular int return-type for mb_strlen. use a more precise type instead.
```

Comment:
it seems `ParametersAcceptorSelector::selectSingle` doesn't pick the return type from the functionMap but from the php8 stubs, where `mb_strlen` is defined with a `int` return type

Reply:
Can you try putting `0|positive-int` return type in `functionMap_php80delta.php` and delete this condition? It might work.

Reply:
ahh that worked - thanks!